src/query/tsdb/remote/server.go

```diff
@@ -113,3 +114,49 @@ func (s *grpcServer) Fetch(
 
 	return err
 }
+
+func (s *grpcServer) Search(
+	message *rpc.SearchRequest,
+	stream rpc.Query_SearchServer,
+) error {
+	var err error
+
+	ctx := RetrieveMetadata(stream.Context())
+	logger := logging.WithContext(ctx)
+	searchQuery, err := DecodeSearchRequest(message)
+	if err != nil {
+		logger.Error("unable to decode search query", zap.Error(err))
+		return err
+	}
+
+	results, err := s.storage.FetchTags(ctx, searchQuery, storage.NewFetchOptions())
+	if err != nil {
+		logger.Error("unable to search tags", zap.Error(err))
+		return err
+	}
+
+	available, pools, err, poolCh, errCh := s.poolWrapper.IteratorPools()
+	if err != nil {
+		return err
+	}
+
+	if !available {
+		select {
+		case pools = <-poolCh:
+		case err = <-errCh:
+			return err
+		}
+	}
+
+	response, err := EncodeToCompressedSearchResult(results, pools)
+	if err != nil {
+		logger.Error("unable to encode search result", zap.Error(err))
```

Comment:
I think you need to `return err` here or else it will try and send a nil message?

Reply:
Sgtm